src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/CitizenSleepHandler.java

```diff
@@ -299,17 +302,30 @@ public float getRenderOffsetZ()
     public boolean shouldGoSleep()
     {
         final BlockPos homePos = findHomePos();
-        final BlockPos citizenPos = citizen.getPosition();
+        BlockPos citizenPos = citizen.getPosition();
+
+        int additionalDist = 0;
+
+        // Additional distance for miners
+        if (citizen.getCitizenData().getJob() instanceof JobMiner && citizen.getCitizenData().getWorkBuilding().getPosition().getY() - 20 > citizenPos.getY())
```

Comment:
Can't we get this directly from the job? So that we create a method in the job that we override? Might scale better

Reply:
ye technically I can add it for all jobs instead, question is if we'd actually use that